tfe/resource_tfe_variable.go

```diff
@@ -0,0 +1,187 @@
+package tfe
+
+import (
+	"fmt"
+	"log"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceTFEVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEVariableCreate,
+		Read:   resourceTFEVariableRead,
+		Update: resourceTFEVariableUpdate,
+		Delete: resourceTFEVariableDelete,
+
+		Schema: map[string]*schema.Schema{
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"value": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"category": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) ([]string, []error) {
+					switch v.(string) {
+					case "env", "terraform":
+						// These are the two valid options, so we return without errors.
+						return nil, nil
+					default:
+						return nil, []error{fmt.Errorf(
+							"%q must be either 'env' or 'terraform'", k),
+						}
+					}
+				},
+			},
+
+			"hcl": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"sensitive": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
```

Comment:
/technically/ we could use this value in a CustomizeDiff function to set the `value` schema field to a sensitive field too, but that may be a bad idea - WDTY @radeksimko?

Reply:
That would maybe be the cleanest solution... But we can also just set the `value` field to be a always be a sensitive field. But indeed curious what @radeksimko thinks about this one? I'm fine with making a custom diff func if you think that is better...